src/hooks/usePayment.jsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const defaultCardInfo = {
+  company: "",
+  number: "",
+  owner: "",
+  expiry: "",
+  nickname: "",
+  cvc: "",
+  password1: "",
+  password2: "",
+  backgroundColor: "#e5e5e5",
+};
+
+const digitTypeInput = ["cvc", "password1", "password2"];
+
+const formatDigitTypeInput = ["number", "expiry"];
+
+const cardCompanyList = [
+  {
+    company: "포코카드",
+    backgroundColor: "red",
+  },
+  {
+    company: "준카드",
+    backgroundColor: "palegreen",
+  },
+  {
+    company: "혜원카드",
+    backgroundColor: "skyblue",
+  },
+  {
+    company: "윤호카드",
+    backgroundColor: "yellow",
+  },
+  // {
+  //   company: "다빈카드",
+  //   backgroundColor: "pink",
+  // },
+  // {
+  //   company: "나나카드",
+  //   backgroundColor: "yellowgreen",
+  // },
+  // {
+  //   company: "치치카드",
+  //   backgroundColor: "plum",
+  // },
+  // {
+  //   company: "사나카드",
+  //   backgroundColor: "lightseagreen",
+  // },
+  // {
+  //   company: "랄랄카드",
+  //   backgroundColor: "burlywood",
+  // },
+];
+
+const usePayment = () => {
+  const [cardInfo, setCardInfo] = useState(defaultCardInfo);
+  const [cardList, setCardList] = useState([]);
+  const [isShowPopup, setIsShowPopup] = useState(true);
+
+  const navigate = useNavigate();
+
+  const onHandleCardInfoInput = evt => {
+    let { value, id } = evt.target;
+    if (!checkValidInputValue(value, id))
+      value = value.subString(0, value.length - 1);
```

Comment:
```suggestion
      value = value.substring(0, value.length - 1);
```
중간에 S가 대문자로 들어가있어서 런타임 에러가 뜨고있어요~